### PR TITLE
refactor: remove `Validation` from `Kinder`

### PR DESCRIPTION
**Pull request by jaschdoc**
This is the final step in removing `Validation` from `Kinder`.

Closes https://github.com/flix/flix/issues/8546

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Kinder.scala [hunk 1]
```diff
@@ -71,18 +70,19 @@ object Kinder {
...
+    val traits = visitTraits(root, taenv, oldRoot, changeSet)
 
-    val defsVal = visitDefs(root, taenv, oldRoot, changeSet)
+    val defs = visitDefs(root, taenv, oldRoot, changeSet)
 
-    val instancesVal = ParOps.parTraverseValues(root.instances)(traverse(_)(i => visitInstance(i, taenv, root)))
+    // Should `instances` not be root.instances.map(ParOps.parMap)?
+    // I.e., for every trait, we check all instances in parallel so we get a parallel task for every instance
+    // instead of check all traits in parallel which spawns k threads each does synchronous code
+    val instances = ParOps.parMapValues(root.instances)(_.map(visitInstance(_, taenv, root)))
```
Comment (jaschdoc) [inline main/src/ca/uwaterloo/flix/language/phase/Kinder.scala:80]: @magnus-madsen see comment above

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Kinder.scala:80]: Not sure. What does @mlutze think?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Kinder.scala:80]: Ill merge this now, but generally avoid comments in code-- better to put them in the PR :)

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Kinder.scala:80]: By comments I mean questions.

We do want comments in the code 🤡

**Review by magnus-madsen — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/language/phase/Kinder.scala (hunk 1 above)
Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/Kinder.scala:80]: Hehe yeah, I was planning to remove that, though... but too late :o

Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/language/phase/Kinder.scala:80]: I can benchmark both versions on the server to check if has an impact

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Kinder.scala:80]: You can make a follow up-PR with your proposed change

**Pull request merged**